core/src/sample_performance_service.rs

```diff
@@ -75,18 +57,12 @@ impl SamplePerformanceService {
 
             if elapsed.as_secs() >= SAMPLE_INTERVAL {
                 now = Instant::now();
-                let (bank, highest_slot) = {
-                    let bank_forks = bank_forks.read().unwrap();
-                    (bank_forks.root_bank(), bank_forks.highest_slot())
-                };
+                let new_snapshot = StatsSnapshot::from_forks(&bank_forks);
 
-                let num_slots = highest_slot.saturating_sub(snapshot.highest_slot);
-                let num_transactions = bank
-                    .transaction_count()
-                    .saturating_sub(snapshot.num_transactions);
-                let num_non_vote_transactions = bank
-                    .non_vote_transaction_count_since_restart()
-                    .saturating_sub(snapshot.num_non_vote_transactions);
+                let (num_transactions, num_non_vote_transactions, num_slots) =
+                    new_snapshot.diff_since(&snapshot);
+                snapshot = new_snapshot;
```

Comment:
Could you add a comment about this line? It's just easy to miss when scanning the code. Maybe also set off with line breaks?
Could steal Steve's verbiage, something like:
Store the new snapshot to compare against in the next iteration of the loop

Reply:
Added a comment and an empty line to create a paragraph.